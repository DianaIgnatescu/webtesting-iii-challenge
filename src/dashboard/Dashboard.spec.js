// Test away
import React from 'react';
import * as rtl from 'react-testing-library';
import 'jest-dom/extend-expect';

import Dashboard from './Dashboard';

afterEach(rtl.cleanup);

describe('Dashboard component', () => {
  it('should render without crashing', () => {
    const wrap = rtl.render(<Dashboard />);
    expect(wrap).toBeTruthy();
    expect(wrap.asFragment()).toMatchSnapshot();
  });
  it('should be able to toggle between open and closed when clicked', () => {
    const wrap = rtl.render(<Dashboard />);
    const closedButton = wrap.getByTestId('closeButton');

    rtl.fireEvent.click(closedButton);
    expect(wrap.getByText('Open Gate'));

    rtl.fireEvent.click(closedButton);
    expect(wrap.getByText('Close Gate'));
  });
  it('should be able to toggle between locked and unlocked when clicked', () => {
    const wrap = rtl.render(<Dashboard />);
    const lockedButton = wrap.getByTestId('lockButton');
    const closedButton = wrap.getByTestId('closeButton');

    rtl.fireEvent.click(closedButton);
    rtl.fireEvent.click(lockedButton);
    expect(wrap.getByText('Unlock Gate'));

    rtl.fireEvent.click(lockedButton);
    expect(wrap.getByText('Lock Gate'));
  });
  it('should not be able to render the locked gate if gate is open', () => {
    const wrap = rtl.render(<Dashboard />);
    const lockedButton = wrap.getByTestId('lockButton');

    rtl.fireEvent.click(lockedButton);
    expect(wrap.queryByText('Unlock Gate')).toBeFalsy();
  })

});
