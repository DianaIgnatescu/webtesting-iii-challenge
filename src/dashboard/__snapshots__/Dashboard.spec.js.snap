// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="lockGate"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closeGate"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="lockButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="closeButton"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
